pennylane/workflow/qnode.py

```diff
@@ -874,73 +957,52 @@ def _execution_component(self, args: tuple, kwargs: dict) -> qml.typing.Result:
             Result
 
         """
-        if (
-            self.device.name == "lightning.qubit"
-            and qml.metric_tensor in self.transform_program
-            and self.diff_method == "best"
-        ):
-            gradient_fn = qml.gradients.param_shift
-        else:
-            gradient_fn = QNode.get_gradient_fn(
-                self.device, self.interface, self.diff_method, tape=self.tape
-            )[0]
-        execute_kwargs = copy.copy(self.execute_kwargs)
-
-        gradient_kwargs = copy.copy(self.gradient_kwargs)
-        if gradient_fn is qml.gradients.param_shift_cv:
-            gradient_kwargs["dev"] = self.device
 
+        execute_kwargs = copy.copy(self.execute_kwargs)
         mcm_config = copy.copy(execute_kwargs["mcm_config"])
-        if not self._tape.shots:
-            mcm_config.postselect_mode = None
-            if mcm_config.mcm_method == "one-shot":
-                raise ValueError(
-                    f"Cannot use the '{mcm_config.mcm_method}' method for mid-circuit measurements with analytic mode."
-                )
 
-        if mcm_config.mcm_method == "single-branch-statistics":
-            raise ValueError("Cannot use mcm_method='single-branch-statistics' without qml.qjit.")
+        config = _make_execution_config(self, self.diff_method, mcm_config=mcm_config)
+
+        # Mapping numpy to None here because `qml.execute` will map None back into
+        # numpy. If we do not do this, numpy will become autograd in `qml.execute`.
+        # If the user specified interface="numpy", it would've already been converted to
+        # "autograd", and it wouldn't be affected.
+        _interface_user_input = None if self.interface == "numpy" else self.interface

```

Comment:
So I think we'll be able to clean this component up if we start using an enum for the interface designation?